### PR TITLE
Systemd files to support starting processes on boot and remote flashing

### PR DESCRIPTION
**Pull request by amrm3lm**
### Description

systemd service files, watcher files, and templated path files that allows us to run programs on boot and restart them when new binaries are received. 

There are two files per Nano service (a .service file and a .path file) and one service file (watcher@) service which is spawned by every .path file.  

### Testing Done

Done on Ubuntu 18 system

### Resolved Issues

#2432 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/jetson_nano/systemd/announcement.service [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# When systemd stops or restarts the thunderbots.service, the action is propagated to this unit
+PartOf=thunderbots.service
+
+# At startup, start this unit after the thunderbots.service start
+After=thunderbots.service thunderloop.service network.target
+
+[Service]
+# Pretend that the component is running
+# todo: execute announcement binary
```
Comment (akhilveeraghanta) [inline src/software/jetson_nano/systemd/announcement.service]: maybe create an overarching ticket here for all of these todos and tag the ticket number

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
nice work on figuring this out! just need a ticket for the TODOs and it LGTM

also needs fix_formatting

**Review by jonathanlew — COMMENTED** [review]
I don't know much about systemd so it'd help to have some documentation around how to use the generic parts

**Review by itsarune — COMMENTED** [review]
Nice! Agree with previous commenters. I wanted to clarify one thing though: systemd is for individual robot management but ansible would be for handling a bunch of robots?

**Comment by amrm3lm** [comment]
> Nice! Agree with previous commenters. I wanted to clarify one thing though: systemd is for individual robot management but ansible would be for handling a bunch of robots?

Yes that is correct. systemd determines what starts at boot and when it restarts. Ansible is just a way of setting up systemd on all robots at the same time (+ being easily adaptable to do anything on all the robots at the same time)

File: src/software/jetson_nano/systemd/announcement.service (hunk 1 above)
Comment (amrm3lm) [reply src/software/jetson_nano/systemd/announcement.service]: Added ticket and fixed formatting

**Review by akhilveeraghanta — APPROVED** [review]
lgtm!

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by itsarune — APPROVED** [review]

**Pull request merged**